File: dotnet/cafdemalihapa/DirectoryOperations.cs
using Microsoft.Extensions.Logging;

namespace cafdemalihapa
{
    public class DirectoryOperations(ILogger<DirectoryOperations> logger)
    {
        public void ListFilesToConsole(string sourceDirectory)
        {
            foreach (var file in Directory.EnumerateFiles(sourceDirectory))
            {
                logger.LogInformation("{file}", file);
            }
        }
    }
}

File: dotnet/cafdemvers/zzq-cafdemwimu/cafdemwimu.console/DirectoryOperations.fs
namespace cafdemwimu.console

open System.IO
open Microsoft.Extensions.Logging

type DirectoryOperations(logger: ILogger<DirectoryOperations>) =
    member _.ListFilesToConsole(sourceDirectory: string) =
        for file in Directory.EnumerateFiles(sourceDirectory) do
            logger.LogInformation("{file}", file)
